### PR TITLE
C++: Add a queries.xml to the test dir

### PR DESCRIPTION
**Pull request by jbj**
This makes compilation caching work with `*.ql` files in the test dir when using `odasa qltest --optimize`. It doesn't break anything in my workflow, and Python already has this. See this Slack discussion: https://semmle.slack.com/archives/C20TR4EAZ/p1549287416378900

### CODE REVIEW

**Comment by rdmarsh2** [comment]
Sounds great to me. I'm going to gibe @geoffw0 and @dave-bartolomeo a chance to weigh in before merging.

**Comment by geoffw0** [comment]
The tests are failing on this.  Seems to be something to do with the test runners path normalization:
```
-| comment_prototypes.c:29:6:29:11 | proto6 | Functions called from other files should be documented (called from $@). | comment_prototypes_caller1.c:9:5:9:10 | call to proto6 | comment_prototypes_caller1.c |
+| comment_prototypes.c:29:6:29:11 | proto6 | Functions called from other files should be documented (called from $@). | comment_prototypes_caller1.c:9:5:9:10 | call to proto6 | query-tests/Documentation/DocumentApi/comment_prototypes_caller1.c |
```

**Comment by jbj** [comment]
I think the test changes are benign. They're essentially changing the output back to what it was before we switched to qltest. See, for example, https://git.semmle.com/Semmle/code/blob/rc/1.17/semmlecode-cpp-tests/library-tests/files/FilePath.expected.

I've added a commit to accept the test changes, plus a workaround for a test that pretends to contain a system header.

File: cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/tainted/queries.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+<queries language="cpp"/>
```
Comment (geoffw0) [inline cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/tainted/queries.xml]: :+1:

**Comment by geoffw0** [comment]
There's now a consistency test failure.  I'm not sure what this means exactly:
```
[706/808] /mnt/resource/jenkins/workspace/Language-Tests/CPP/ql/cpp/ql/test/query-tests/Security/CWE/CWE-190/semmle/tainted: semmlecode-cpp-consistency-queries/unusedLocations.ql: FAILED (compilation: 47ms, execution: 26ms, total: 87ms)
Consistency test has output:
| queries.xml:1:1:1:26 | queries.xml:1:1:1:26 |
| queries.xml:1:1:1:26 | queries.xml:1:1:1:26 |
```

**Comment by jbj** [comment]
@ian-semmle Is that a matter of adding another line to https://git.semmle.com/Semmle/code/blob/master/semmlecode-cpp-consistency-queries/unusedLocations.ql, exempting locations in XML files? I'd also like to hear if you have further input on the approach I'm taking with this PR.

**Comment by ian-semmle** [comment]
I've opened an internal PR to teach unusedLocations.ql about XML files. The approach generally seems good to me.

**Comment by geoffw0** [comment]
@jbj have you confirmed the fix works?  Then once it's merged I'll be happy to merge this.

**Comment by jbj** [comment]
@geoffw0 I've merged Ian's fix in the internal repo and retriggered the tests here, and they are now passing. That means this PR is safe to merge.

**Pull request merged**